lib/amorail/entity/persistance.rb

```diff
@@ -40,5 +40,21 @@ def reload
     def extract_data_add(response)
       response.fetch('add').first
     end
+
+    def extract_data_update(response)
+      case data = response.fetch('update')
+      when Array
+        data.first
+      when Hash
+        merge_errors(data)
+        raise(InvalidRecord)
+      end
+    end
+
+    def merge_errors(data)
+      data["errors"].each do |_, message|
```

Comment:
Let's use `fetch` here then to have a more meaningful error message if "errors" key is missing somehow.